hideShowPassword.js

```diff
@@ -314,10 +315,17 @@
         $.each(options.inheritStyles, $.proxy(function (index, prop) {
           options.styles[prop] = this.element.css(prop);
         }, this));
-        this.element.css(options.innerElementStyles).wrap(
-          $(options.element).addClass(options.className).css(options.styles)
-        );
-        this.wrapperElement = this.element.parent();
+        this.element.css(options.innerElementStyles);
+        if (!options.element.jquery) {
```

Comment:
jQuery elements are also valid jQuery selectors. So instead of checking if the selector is a jQuery object, we could instead check to make sure it's not already the parent element before wrapping.

Example:

```js
// Keep this separate, as you've already done
this.element.css(options.innerElementStyles);
// Create the wrapper element, using `options.element`
this.wrapperElement = $(options.element).addClass(options.className).css(options.styles);
// If this wrapper element doesn't already have the element as a descendent...
if (!this.wrapperElement.has(this.element).length) {
  // ...then wrap the descendent and override the local reference
  this.element.wrap(this.wrapperElement);
  this.wrapperElement = this.element.parent();
}
```